UI/Controllers/CandidatesController.cs

```diff
@@ -0,0 +1,49 @@
+﻿using BLL.DTO;
+using DAL;
+using DAL.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DAL.Repositories;
+using Microsoft.AspNetCore.Routing.Matching;
+using BLL.Interfaces;
+using BLL.Services;
+
+namespace UI.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class CandidatesController : ControllerBase
+    {
+        private ICandidateService candidateService;
```

Comment:
private readonly

Reply:
where is the constructor with service injection? Do you test this code? This code doesn't work!